riverdriver/riverpgxv5/river_pgx_v5_driver.go

```diff
@@ -278,6 +279,108 @@ func (e *Executor) JobInsertFull(ctx context.Context, params *riverdriver.JobIns
 	return jobRowFromInternal(job)
 }
 
+const jobInsertManyReturningTemplate = `-- name: JobInsertManyReturningTemplate :execrows
+INSERT INTO river_job(
+    args,
+    kind,
+    max_attempts,
+    metadata,
+    priority,
+    queue,
+    scheduled_at,
+    state,
+    tags
+) VALUES
+%s
+RETURNING id, args, attempt, attempted_at, attempted_by, created_at, errors, finalized_at, kind, max_attempts, metadata, priority, queue, state, scheduled_at, tags, unique_key
+`
+
+func (e *Executor) JobInsertManyReturning(ctx context.Context, params []*riverdriver.JobInsertFastParams) ([]*rivertype.JobRow, error) {
```

Comment:
So I went to deduplicate these between drivers as they're nearly the same other than the function name called to query and the return value scanning. Even just doing the sql building part would be nice.

What I ran into though was it felt difficult to do it without causing lots of extra copying and allocation. Can maybe take another pass at it later unless you have ideas.

Reply:
Irrelevant now that the implementation doesn't use the query string builder approach.